gammapy/data/data_store.py

```diff
@@ -412,6 +474,29 @@ def get_observations(
         log.info(f"Observations selected: {len(obs_list)} out of {len(obs_id)}.")
         return Observations(obs_list)
 
+    def get_observation_groups(self, key):
+        """Generate groups of `~gammapy.data.Observations` with a shared property.
+
+        Parameters
+        ----------
+        key : str
+            Key of the observation table used to apply the grouping.
+            For example "EVENT_TYPE" will return group observations
+            with the same event type.
+
+        Returns
+        -------
+        groups : dict of `~gammapy.data.Observations`
+            Dictionary of Observations instance, one instance for each group.
+        """
+
+        observations = self.get_observations()
+        observations_group = observations.group_by_label(self.obs_table[key])
+        reformated = dict()
+        for old_key, value in observations_group.items():
+            reformated[f"{key}{old_key[5:]}"] = value
+        return reformated
```

Comment:
```suggestion
       return {
           f"{key}{old_key[5:]}":  value
           for old_key, value in observations_group.items()
       } 
```
should be faster than an explicit `for-loop`